soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/rollup/RollupExact.scala

```diff
@@ -830,9 +830,10 @@ class RollupExact[MT <: MetaTypes](
         case Some((selectedColumn, NamedExpr(_ : WindowedFunctionCall, _name, _hint, _isSynthetic), _)) if rollupContext.isCoarseningGroup =>
           log.debug("can't rewrite, windowed function call")
           None
-        case Some((selectedColumn, ne, functionExtract)) =>
-          assert(ne.expr.typ == sExpr.typ)
-          Some(functionExtract(PhysicalColumn[MT](newFrom.label, newFrom.tableName, columnLabelMap(selectedColumn), sExpr.typ)(sExpr.position.asAtomic)))
+        case Some((selectedColumn, rollupExpr, functionExtract)) =>
+          val result = functionExtract(PhysicalColumn[MT](newFrom.label, newFrom.tableName, columnLabelMap(selectedColumn), rollupExpr.typ)(sExpr.position.asAtomic))
+          assert(result.typ == sExpr.typ)
+          Some(result)
```

Comment:
This wasn't quite asserting on the right thing; before "functionExtract" never changed the type of the thing it was extracting from.  Now it can, so we assert that the result of the extraction is the same as the type of the source expr.